OSDOCS-5543:Adding performance and resource recommendatiaons (#57451)
openshift/openshift-docs: modules/network-observability-resources-table.adoc

//module included in the following assemblies:

> // network_observability/configuring_operator.adoc

Reviewer: ```suggestion
// network_observability/configuring_operator.adoc
```

More of a nit than anything, but add a space after the commented out clauses with the doc's structural line entries. More for consistency with other modules.

Reviewer 2: A really nitty nit, should be // Module

Author: Good ones, TY!

:_content-type: REFERENCE
[id="network-observability-resources-table{context}"]

> = Resource recommendations

Reviewer: Resource considerations?

> The following table documents recommendations for resources depending on workload size. 

Reviewer: ```suggestion
The following table outlines examples of resource settings for clusters with a certain node capacities.
```

Author: Addressed, but I think I will leave workload size.

[IMPORTANT]
====

> These recommended scenarios are tailored to specific workloads and should serve only as a baseline from which adjustments can be made to accommodate your workload needs.

Reviewer: ```suggestion
The examples outlined in the table, demonstrate scenarios that are tailored to specific workloads. Consider each example as a baseline from which adjustments can be made to accommodate your workload needs.
```

Author: Addressed

====

.Resource recommendations
[options="header"]
|============================================================================================================================
|                                     | Extra small (10 Nodes) | Small (25 nodes)  | Medium (65 Nodes) ^[2]^ | Large (120 nodes) ^[2]^
| *Worker Node vCPU and Memory*       | 4 vCPUs\| 16GiB Mem ^[1]^ | 16 vCPUs\| 64GiB Mem ^[1]^ | 16 vCPUs\| 64GiB Mem  ^[1]^  |16 vCPUs\| 64GiB Mem ^[1]^ 

> | *LokiStack Size*                    | 1x.extra-small         | 1x.small          | 1x.small           | 1x.medium

Reviewer: `extra-small|small|small|medium` correct? No "large"? Just want to check because of the column headings.

Reviewer: Sorry, I should have clarified: Does the LokiStack Size "1x.medium` entry definitely sit with the "Large (120 nodes) column" of would a LokiStack Size "1x.mlarge` entry (if it exists), be a better fit?

Author: Yes, the Large (120 nodes) correlates with the LokiStack size `1.xmedium`. This is based on QE performance testing data, you can see here: https://docs.google.com/document/d/1dKfMfHCtvXAh1-QE9vk08e3DWBjo7t2BSuNMcE1ECTQ/edit

Reviewer: Thanks, Sarah. That makes sense. Feel free to mark this thread as Resolved.

| *NetObserv Controller Memory Limit* | 400Mi (default)        | 400Mi (default)   | 400Mi (default)    | 800Mi                
| *eBPF Sampling Rate*                | 50 (default)           | 50 (default)      | 50 (default)       | 50 (default)
| *eBPF Memory Limit*                 | 800Mi (default)        | 800Mi (default)   | 2000Mi             | 800Mi (default) 
| *FLP Mem Limit*                     | 800Mi (default)        | 800Mi (default)   | 800Mi (default)    | 800Mi (default)         
| *FLP Kafka Partitions*              | N/A                    | 48                | 48                 | 48            
| *Kafka Consumer Replicas*           | N/A                    | 24                | 24                 | 24
| *Kafka Brokers*                     | N/A                    | 3 (default)       | 3 (default)        | 3 (default)
|============================================================================================================================

> 1. Tested with AWS m6i images.

Reviewer 3: ```suggestion
1. Tested with AWS m6i instances.
```

> 1. Tested with AWS m6i instances.

Reviewer: M6i? as per https://aws.amazon.com/about-aws/whats-new/2021/08/amazon-ec2-m6i-instances/

Author: Ah yep, that's right.

> 2. In addition to this worker and its controller, 3 infra nodes (size m6.12xlarge) and 1 workload node (size m6.8xlarge) were tested. 

Reviewer 3: ```suggestion
2. In addition to this worker and its controller, 3 infra nodes (size m6i.12xlarge) and 1 workload node (size m6i.8xlarge) were tested. 
```

> 2. In addition to this worker and its controller, 3 infra nodes (size m6i.12xlarge) and 1 workload node (size m6i.8xlarge) were tested. 

Reviewer: ```suggestion
2. In addition to this worker and its controller, 3 infra nodes (size `m6i.12xlarge`) and 1 workload node (size `m6i.8xlarge`) were tested. 
```

// Module included in the following assemblies:

> // network_observability/configuring_operator.adoc

Author: I did this but didn't add the bullet

:_content-type: REFERENCE
[id="network-observability-resources-table{context}"]
= Resource considerations
The following table outlines examples of resource considerations for clusters with certain workload sizes. 

[IMPORTANT]
====
The examples outlined in the table demonstrate scenarios that are tailored to specific workloads. Consider each example only as a baseline from which adjustments can be made to accommodate your workload needs.
====

.Resource recommendations
[options="header"]
|============================================================================================================================

> |                                     | Extra small (10 Nodes) | Small (25 nodes)  | Medium (65 Nodes) ^[2]^ | Large (120 nodes) ^[2]^
> | *Worker Node vCPU and Memory*       | 4 vCPUs\| 16GiB Mem ^[1]^ | 16 vCPUs\| 64GiB Mem ^[1]^ | 16 vCPUs\| 64GiB Mem  ^[1]^  |16 vCPUs\| 64GiB Mem ^[1]^ 
> | *LokiStack Size*                    | 1x.extra-small         | 1x.small          | 1x.small           | 1x.medium
> | *NetObserv Controller Memory Limit* | 400Mi (default)        | 400Mi (default)   | 400Mi (default)    | 800Mi                
> | *eBPF Sampling Rate*                | 50 (default)           | 50 (default)      | 50 (default)       | 50 (default)
> | *eBPF Memory Limit*                 | 800Mi (default)        | 800Mi (default)   | 2000Mi             | 800Mi (default) 
> | *FLP Mem Limit*                     | 800Mi (default)        | 800Mi (default)   | 800Mi (default)    | 800Mi (default)         
> | *FLP Kafka Partitions*              | N/A                    | 48                | 48                 | 48            
> | *Kafka Consumer Replicas*           | N/A                    | 24                | 24                 | 24
> | *Kafka Brokers*                     | N/A                    | 3 (default)       | 3 (default)        | 3 (default)

Reviewer 4: Nit: Usually headers (both the top actual header and the manual left-side one) should be in sentence case. Idk if the following would be fully accurate, but potentially something like:

```suggestion
|                                     | Extra small (10 nodes) | Small (25 nodes)  | Medium (65 nodes) ^[2]^ | Large (120 nodes) ^[2]^
| *Worker node vCPU and memory*       | 4 vCPUs\| 16GiB mem ^[1]^ | 16 vCPUs\| 64GiB mem ^[1]^ | 16 vCPUs\| 64GiB mem  ^[1]^  |16 vCPUs\| 64GiB mem ^[1]^ 
| *LokiStack size*                    | `1x.extra-small`         | `1x.small`          | `1x.small`           | `1x.medium`
| *NetObserv controller memory limit* | 400Mi (default)        | 400Mi (default)   | 400Mi (default)    | 800Mi                
| *eBPF sampling rate*                | 50 (default)           | 50 (default)      | 50 (default)       | 50 (default)
| *eBPF memory limit*                 | 800Mi (default)        | 800Mi (default)   | 2000Mi             | 800Mi (default) 
| *FLP memory limit*                     | 800Mi (default)        | 800Mi (default)   | 800Mi (default)    | 800Mi (default)         
| *FLP Kafka partitions*              | N/A                    | 48                | 48                 | 48            
| *Kafka consumer replicas*           | N/A                    | 24                | 24                 | 24
| *Kafka brokers*                     | N/A                    | 3 (default)       | 3 (default)        | 3 (default)
```

Also I noticed in the footnotes after this table that the sizes were marked up (e.g.,  `M6i.12xlarge`), so I showed marking them up in the above as well, in case that's valid.

Author: Good suggestions thank you! Addressed.

|============================================================================================================================

> 1. Tested with AWS M6i instances.
> 2. In addition to this worker and its controller, 3 infra nodes (size `M6i.12xlarge`) and 1 workload node (size `M6i.8xlarge`) were tested. 

Reviewer 4: Suggestion: When using the `^[1]^` style of table-footnotes mentioned in the repo guidelines ([Alternative footnote styling in tables](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#alternative-footnote-styling-in-tables)), I'd suggest using the `[.small]` style shown there, because it makes it show up more like what footnotes normally look like:

```suggestion
[.small]
--
1. Tested with AWS M6i instances.
2. In addition to this worker and its controller, 3 infra nodes (size `M6i.12xlarge`) and 1 workload node (size `M6i.8xlarge`) were tested. 
--
```

Author: Niiice, thanks for this suggestion :) Addressed.
